app/controllers/dragons_teams_controller.rb

```diff
@@ -20,8 +20,10 @@ def create
     if current_user.dragons.count >= 5
       flash[:alert] = "You can't add more dragons"
       redirect_to dragons_teams_index_path(current_user.id)
-    else
+    elsif can_afford
```

Comment:
Metody `can_afford` oraz `pay_for_dragon` to logika biznesowa która nie powinna być zaimplementowana w kontrolerze. Dobrze byłoby je wyciągnąć do osobnej klasy.

`can_afford`, możemy przenieść do modelu User

czyli tutaj mielibyśmy zamiast `elsif can_afford`
```
elsif current_user.can_afford?(DragonType.find(params[:dragons_team][:dragon_type_id])
```

Niestety w modelu nie możemy ustawić flash[:alert], więc trzeba by to rozwiązać tak:
```ruby
dragon_type = DragonType.find(params[:dragons_team][:dragon_type_id]
if current_user.dragons.count >= 5
  flash[:alert] = "You can't add more dragons"
  redirect_to dragons_teams_index_path(current_user.id)
elsif !current_user.can_afford?(dragon_type) # (z negacją)
  flash[:alert] = current_user.missing_resources_for(dragon_type)
```

gdzie `missing_resources_for` jest kolejną metodą zaimplementowaną w modelu User.

Wtedy mamy bardziej klarowne warstwy aplikacji (kontroler nie "myśli" tylko przekazuje dalej dane) i fajne, obiektowe podejście (do modelu user przekazujemy obiekt DragonType i to obiekt user pytamy czy go stać, a nie kontroler który jest tylko abstrakcją do odbierania zapytań z przeglądarki)